cmd/state-offline-uninstaller/uninstall.go

```diff
@@ -54,11 +67,48 @@ func newParams() *Params {
 	return &Params{path: "/tmp"}
 }
 
+func (r *runner) Event(eventType string, installerDimensions *dimensions.Values) {
+	r.analytics.Event(ac.CatRuntimeUsage, eventType, installerDimensions)
+}
+
+func (r *runner) EventWithLabel(eventType string, msg string, installerDimensions *dimensions.Values) {
+	r.analytics.EventWithLabel(ac.CatRuntimeUsage, eventType, msg, installerDimensions)
+}
+
+func (r *runner) handleFailure(err error, msg string, installerDimensions *dimensions.Values) error {
+	r.EventWithLabel("failure", msg, installerDimensions)
+	return errs.Wrap(err, msg)
+}
+
+func (r *runner) handleFailureNewErr(msg string, installerDimensions *dimensions.Values) error {
+	r.EventWithLabel("failure", msg, installerDimensions)
```

Comment:
I'm not a fan of submitting failure messages to analytics, much less to the heartbeat category. These should be reported to rollbar. If we also want to surface these in our metrics then I'd like to see a use-case for it.

The runtime-use category is stable, so we should avoid pushing anything new to it that's experimental or without a strong use-case.

If your concern is that you might loose out on data, or that you want this data in the data warehouse, then I would recommend using a new category that we can mark as unstable for the time being.

Reply:
Offline installer now has its own category.  We expect the installers to fail a lot and want to make it trivial for DW to gather statistics on failure modes in Athena